server/src/main/java/io/unitycatalog/server/persist/CommitRepository.java

```diff
@@ -107,4 +111,24 @@ public void updateTableMetadata(Session session, Commit commit) {
     tableInfoDAO.setColumns(ColumnInfoDAO.fromList(metadata.getSchema().getColumns()));
     session.merge(tableInfoDAO);
   }
+
+  public List<CommitDAO> getLatestCommits(UUID tableId) {
+    try (Session session = SESSION_FACTORY.openSession()) {
+      Transaction tx = session.beginTransaction();
+      try {
```

Comment:
Do we need transaction for a single query?

Reply:
removed